sweepbatcher/sweep_batch.go

```diff
@@ -807,6 +812,24 @@ func (b *batch) sweepExists(outpoint wire.OutPoint) bool {
 	return ok
 }
 
+// totalValue returns the sum of all sweep values in the batch.
+func (b *batch) totalValue() btcutil.Amount {
+	var total btcutil.Amount
+	for _, sweep := range b.sweeps {
+		total += sweep.value
+	}
+	return total
+}
+
+// exceedsThreshold checks if the batch value exceeds the immediate publish
+// threshold. Returns false if the threshold is not configured (0).
```

Comment:
I propose to add the amount as an argument or to add a note that the method must be called from the batch's event loop.